src/discover/learner.js

```diff
@@ -3,21 +3,26 @@ const Learner = require('ac-learn')
 
 const JSON_PATH = `${__dirname}/learner.json`
 
-//@TODO: Use the JSON methods from `ac-learn` to get the whole thing saveable
-const learner = new Learner()
-/* eslint-disable no-console */
-if (existsSync(JSON_PATH)) {
-  learner.loadAndDeserializeClassifier(JSON_PATH).then(classifier => {
-    learner.classifier = classifier
-    // console.log('Re-using existing classifier')
-  }, console.error)
-} else {
-  learner.crossValidate(6)
-  learner.eval()
-  learner.serializeAndSaveClassifier(JSON_PATH).then(_ => {
-    // console.log('Classifier saved', classifier)
-  }, console.error)
+async function getLearner() {
+  const learner = new Learner()
+
+  try {
+    if (existsSync(JSON_PATH)) {
+      learner.classifier = await learner.loadAndDeserializeClassifier(JSON_PATH)
+    } else {
+      learner.crossValidate(6)
+      learner.eval()
+
+      await learner.serializeAndSaveClassifier(JSON_PATH)
+    }
+  } catch (e) {
+    /* eslint-disable no-console */
+    console.error(e)
+  }
+
+  return learner
 }
-/* eslint-enable no-console */
 
-module.exports = learner
+module.exports = {
+  getLearner,
```

Comment:
Not sure what could be exported, but moving to a named export allows for a smaller change in the future if something is added. Ie. only one line will change, instead of three. This also ensures consistency in import names where default imports can be renamed easily making readability a little worse if youre debugging and trying to trace code.

I removed the learner.json file since it was empty. According to [this](https://www.npmjs.com/package/ac-learn#serializeandsaveclassifier), the file will be generated if it doesn't exist. No point in commiting an empty file.

Reply:
> Not sure what could be exported, but moving to a named export allows for a smaller change in the future if something is added. Ie. only one line will change, instead of three. This also ensures consistency in import names where default imports can be renamed easily making readability a little worse if youre debugging and trying to trace code.

That's a great point.

> I removed the learner.json file since it was empty. According to this, the file will be generated if it doesn't exist. No point in commiting an empty file.

Hmm, that shouldn't be the case, it should have had the latest model I got using `ac-learn`.
I wonder how it ended up being empty. I'll look into it that either this early morning later (unless you want to do that?).